test/external/external_test_opt.py

```diff
@@ -215,7 +215,7 @@ def test_fold_4convs_sgd(self):
       c3 = nn.Conv2d(8,16,3,bias=False)
       c4 = nn.Conv2d(16,32,3,bias=False)
       opt = optim.SGD(get_parameters([c1, c2, c3, c4]))
-      with CLCache(allowed=19):
+      with CLCache(allowed=22):
```

Comment:
Why is this increasing?

Reply:
Reshape which is pushed up is getting stuck b/w ReduceOps.SUM & BinaryOps.MAX. Because ReduceOp LB has 2 , reshape can't be pushed up more.  Hence Binary & Reduce are not getting fused.

I think a better solution of handling reshapes through BinOp can exist,. I will think more of what condition can be imposed more.
![image](https://github.com/tinygrad/tinygrad/assets/88420255/ead739e0-fa27-4388-a67f-83446ad51964)

Reply:
test_fold_2convs & 4convs have the same above problem

Reply:
This needs to be fixed